packages/react/src/components/UIShell/HeaderContainer.tsx

```diff
@@ -26,17 +38,13 @@ const HeaderContainer = ({ isSideNavExpanded, render: Children }) => {
       onClickSideNavExpand={handleHeaderMenuButtonClick}
     />
   );
-};
+}
 
 HeaderContainer.propTypes = {
   /**
    * Optionally provide a custom class name that is applied to the underlying <header>
    */
   isSideNavExpanded: PropTypes.bool,
-};
 
-HeaderContainer.defaultProps = {
-  isSideNavExpanded: false,
+  render: PropTypes.elementType,
```

Comment:
Should this be `PropTypes.func`? We should also add a description of the prop if we are adding it to the `propType` definitions

Reply:
I added a description with an example. Should doc comments be added to the HeaderContainerProps interface as well? When I use the element in this repo vscode displays the doc comments from the propTypes for it, but I'm wondering if it will do that when this package is used

Also, I believe that elementType is the correct type for the render prop because the way it's used within the HeaderContainer prop is as an element:
https://github.com/carbon-design-system/carbon/blob/b0a17c4e5db755184763eec47391628ff6fd0a34/packages/react/src/components/UIShell/HeaderContainer.tsx#L35-L40
A function that returns a JSX.Element is a component

Reply:
> Should doc comments be added to the HeaderContainerProps interface as well?

I don't believe we've been adding with the other typescript refactors


> A function that returns a JSX.Element is a component

Thanks for clarifying! 👍🏻